### ISSUE TITLE
Segnet+Cityscapes : Accuracy issue

### ISSUE THREAD

**Issue by athus1990** [alexgkendall/SegNet-Tutorial#39]
Hi,
I trained segnet(non basic non Bayesian version ) on the Camvid dataset(44gb of 19k+ images and annotations that need to be converted to segnet annotations). However training it on cityscapes and testing it on a new test image does not produce the same "smooth"  results as shown by webcam demo.
I tried to show the difference here(excuse the color difference:red is road and purple is pavement and so on:

My soln(the road and pavements are not smooth,many miscalssifications):
![image](https://cloud.githubusercontent.com/assets/7275643/14894000/85216826-0d3f-11e6-8468-95e3e689261a.png)

Your webdemo:
![image](https://cloud.githubusercontent.com/assets/7275643/14894114/1d8a4196-0d40-11e6-89ba-a617c23d5b13.png)

However on camvid testimage:
![image](https://cloud.githubusercontent.com/assets/7275643/14894157/591bfefc-0d40-11e6-84fa-0b31fa415c2e.png)

So here goes my question:
1)webdemo was trained on ~500 images it gives great results.I trained segnet with 19k+ images and 90000+ iterations and dint change anything else.Why am I not getting a smooth performance. Should it not it do better. 
2)Also the cityscapes dataset has a resolution of 1024x2048 which i rescaled to 360x480 for segnet. My test image which segnet has never trained on is 1080x1920 which i rescaled to 360x480.Does resolution matter?
3)Are testing it differently to get a smoother performance.?All I am doing is similar to testcamvid.py in ur code:

image= cv2.resize(image, (input_shape[3], input_shape[2]))
input_image = image.transpose((2, 0, 1))
input_image = input_image[(2, 1, 0), :, :]
input_image = np.asarray([input_image])

out = net.forward_all(data=input_image)
segmentation_ind = np.squeeze(net.blobs['argmax'].data)
segmentation_ind_3ch = np.resize(segmentation_ind, (3, input_shape[2], input_shape[3]))
segmentation_ind_3ch = segmentation_ind_3ch.transpose(1, 2, 0).astype(np.uint8)
segmentation_rgb = np.zeros(segmentation_ind_3ch.shape, dtype=np.uint8)
cv2.LUT(segmentation_ind_3ch, label_colours, segmentation_rgb)
segmentation_rgb = segmentation_rgb.astype(float) / 255.0 

Could you please help me understand this better?

**Comment by Timo-hab** [comment]
Are you progressed? Could you successfully train and test SegNet on Cityscapes database?

**Comment by athus1990** [comment]
Yea.But it is not as great as training with camvid.

**Comment by varunmurali1** [comment]
Did you actually get reasonable results from training. I am still getting an output which is majority of one label. How did you run the training? Did you just fine-tune from the VGG net?

**Comment by athus1990** [comment]
yes the results were great! Just not great enough.I did start with Vgg16. Converted cityscapes Annotations-->similar to Camvids and then changed the number of classes.

**Comment by varunmurali1** [comment]
Could you share the trained model on the Cityscapes dataset for evaluation purposes?

**Comment by athus1990** [comment]
Sure.just give me an hour

**Comment by varunmurali1** [comment]
Great!

**Comment by varunmurali1** [comment]
Hey @athus1990 , would it be possible to share the model?

**Comment by acgtyrant** [comment]
This issue can be closed now...

**Comment by saviogeorge** [comment]
Hi athus1990,

I was trying to train the segnet with the cityscapes data set however i get the following error
 Check failed: status == CUBLAS_STATUS_SUCCESS (11 vs. 0)  CUBLAS_STATUS_MAPPING_ERROR
I believe this is due to the difference in annotations used by the city scapes data set.
![annotations_problem](https://user-images.githubusercontent.com/20711873/28362455-bf42be40-6c7c-11e7-93a0-479e5af116c2.png)
Here you have said you have converted the cityscapes Annotations-->similar to Camvids.
Can you please let me know how did you do it. some script?

**Comment by guanfuchen** [comment]
@athus1990 what is the mIoU about your model trained on Cityscapes, the loss does not seem to fall when the iterations increase.
